examples/src/components/utils/ExampleWithExport.jsx

```diff
@@ -0,0 +1,148 @@
+// (C) 2007-2018 GoodData Corporation
+/* eslint-disable react/jsx-closing-tag-location */
+import React from 'react';
+import assign from 'lodash/assign';
+import get from 'lodash/get';
+import ExportDialog from '@gooddata/goodstrap/lib/Dialog/ExportDialog';
+import PropTypes from 'prop-types';
+
+export class ExampleWithExport extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            showExportDialog: false,
+            errorMessage: null
+        };
+
+        this.onExportReady = this.onExportReady.bind(this);
+        this.exportToCSV = this.exportToCSV.bind(this);
+        this.exportToXLSX = this.exportToXLSX.bind(this);
+        this.exportWithCustomName = this.exportWithCustomName.bind(this);
+        this.exportWithDialog = this.exportWithDialog.bind(this);
+        this.exportWithMergeHeaders = this.exportWithMergeHeaders.bind(this);
+        this.doExport = this.doExport.bind(this);
+        this.getExportDialog = this.getExportDialog.bind(this);
+        this.exportDialogCancel = this.exportDialogCancel.bind(this);
+    }
+
+    onExportReady(exportResult) {
+        this.exportResult = exportResult;
+    }
+
+    getExportDialog() {
+        return (
+            <ExportDialog
+                headline="Export to XLSX"
+                cancelButtonText="Cancel"
+                submitButtonText="Export"
+                isPositive
+                seleniumClass="s-dialog"
+
+                mergeHeaders
+                mergeHeadersDisabled={false}
+                mergeHeadersText="Keep attribute cells merged"
+                mergeHeadersTitle="CELLS"
+
+                onCancel={this.exportDialogCancel}
+                onSubmit={this.exportWithMergeHeaders}
+            />
+        );
+    }
+
+    downloadFile(uri) {
+        let anchor = document.getElementById('downloader');
+        if (!anchor) {
+            anchor = document.createElement('a');
+            anchor.id = 'downloader';
+            document.body.appendChild(anchor);
+        }
+        anchor.href = uri;
+        anchor.download = uri;
+        anchor.click();
+    }
+
+    exportDialogCancel() {
+        this.setState({ showExportDialog: false });
+    }
+
+    exportToCSV() {
+        this.doExport({});
+    }
+
+    exportToXLSX() {
+        this.doExport({ format: 'xlsx' });
+    }
+
+    exportWithCustomName() {
+        this.doExport({ title: 'CustomName' });
+    }
+
+    exportWithDialog() {
+        this.setState({ showExportDialog: true });
+    }
+
+    exportWithMergeHeaders(exportConfig) {
+        this.setState({ showExportDialog: false });
```

Comment:
why only this method closes opened export dialog?

Reply:
Only exportWithDialog method will show ExportDialog. 
The other export buttons will export CSV (no ExportDialog).

Reply:
I understand, but other functions like exportToXLSX() call just odExport, only this method also sets state even that also exportWithMergeHeaders doesn't show exportDialog

Reply:
The flow is:
exportWithDialog => show ExportDialog => submit => exportWithMergeHeaders => hide ExportDialog + doExport

exportWithMergeHeaders is called by ExportDialog only

Reply:
ok, so why it is not used in this way in storybook? In storybook Export with mergeHeaders button doesn't show export dialog

Reply:
I think storybook is used for simple cases. So I didn't add ExportDialog into it.
I will do it now